src/webgpu/api/validation/createComputePipeline.spec.ts

```diff
@@ -0,0 +1,164 @@
+  }
+
+  getInvalidShaderModule(): GPUShaderModule {
+    this.device.pushErrorScope('validation');
+    const code = 'deadbeaf'; // Something make nonsense
+    const shaderModule = this.device.createShaderModule({ code });
+    this.device.popErrorScope();
+    return shaderModule;
+  }
+
+  doCreateComputePipelineTest(
+    isAsync: boolean,
+    _success: boolean,
+    descriptor: GPUComputePipelineDescriptor
+  ) {
+    if (isAsync) {
+      if (_success) {
+        this.shouldResolve(this.device.createComputePipelineAsync(descriptor));
+      } else {
+        this.shouldReject('OperationError', this.device.createComputePipelineAsync(descriptor));
+      }
+    } else {
+      if (_success) {
+        this.device.createComputePipeline(descriptor);
+      } else {
+        this.expectValidationError(() => {
+          this.device.createComputePipeline(descriptor);
+        });
+      }
+    }
+  }
+}
+
+export const g = makeTestGroup(F);
+
+g.test('basic_use_of_createComputePipeline')
+  .cases(poptions('isAsync', [true, false]))
+  .fn(async t => {
+    const { isAsync } = t.params;
+    t.doCreateComputePipelineTest(isAsync, true, {
+      compute: { module: t.getShaderModule('compute', 'main'), entryPoint: 'main' },
+    });
+  });
+
+g.test('shader_module_must_be_valid')
+  .cases(poptions('isAsync', [true, false]))
+  .fn(async t => {
+    const { isAsync } = t.params;
+    t.doCreateComputePipelineTest(isAsync, false, {
+      compute: {
+        module: t.getInvalidShaderModule(),
+        entryPoint: 'main',
+      },
+    });
+  });
+
+g.test('shader_module_stage_must_be_compute')
+  .cases(
+    params()
+      .combine(poptions('isAsync', [true, false]))
+      .combine(
+        poptions('shaderModuleStage', [
+          'compute' as TShaderStage,
+          'vertex' as TShaderStage,
+          'fragment' as TShaderStage,
+        ])
+      )
+  )
+  .fn(async t => {
+    const { isAsync, shaderModuleStage } = t.params;
+    const descriptor = {
+      compute: {
+        module: t.getShaderModule(shaderModuleStage, 'main'),
+        entryPoint: 'main',
+      },
+    };
+    t.doCreateComputePipelineTest(isAsync, shaderModuleStage === 'compute', descriptor);
+  });
+
+g.test('enrty_point_name_must_match')
+  .cases(
+    params()
+      .combine(poptions('isAsync', [true, false]))
+      .combine([
+        { shaderModuleEntryPoint: 'main', stageEntryPoint: 'main' },
+        { shaderModuleEntryPoint: 'main', stageEntryPoint: '' },
+        { shaderModuleEntryPoint: 'main', stageEntryPoint: 'main\0' },
+        { shaderModuleEntryPoint: 'main', stageEntryPoint: 'main\0a' },
+        { shaderModuleEntryPoint: 'main', stageEntryPoint: 'mian' },
+        { shaderModuleEntryPoint: 'main', stageEntryPoint: 'main ' },
+        { shaderModuleEntryPoint: 'main', stageEntryPoint: 'main\n' },
+        { shaderModuleEntryPoint: 'mian', stageEntryPoint: 'mian' },
+        { shaderModuleEntryPoint: 'mian', stageEntryPoint: 'main' },
```

Comment:
Could we also have a couple tests with `shaderModuleEntryPoint` != "main"? I'm asking because for a very long time Dawn didn't support entrypoints other than "main".

Reply:
In fact the last two tests have shaderModuleEntryPoint === "mian" rather than "main". Maybe I can add more cases like "mainmain" or "entryPoint0"?

Reply:
Great!